src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,460 @@
+using Microsoft.VisualStudio.LanguageServices.Setup;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// Interaction logic for SampleToolboxUserControl.xaml
+    /// </summary>
+    internal partial class SampleToolboxUserControl : UserControl, IOleCommandTarget
+    {
+        public SampleToolboxUserControl(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            InitializeComponent();
+            InitializeIfNeeded(workspace, documentTrackingService, languageServiceBroker, threadingContext);
+        }
+
+        private Workspace? workspace { get; set; }
+        private DocumentId? lastDocumentId { get; set; }
+        private List<DocSymbol>? originalTree { get; set; }
+        private ITextSnapshot? snapshot { get; set; }
+        private IWpfTextView? textView { get; set; }
+
+        private void InitializeIfNeeded(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            this.workspace = workspace;
+            documentTrackingService.ActiveDocumentChanged += DocumentTrackingService_ActiveDocumentChanged;
+
+            this.textView = GetActiveTextView();
+            if (this.textView is not null)
+            {
+                this.textView.Caret.PositionChanged += FollowCursor;
+            }
+
+            searchBox.GotFocus += RemoveText;
+            searchBox.LostFocus += AddText;
+
+            async void DocumentTrackingService_ActiveDocumentChanged(object sender, DocumentId? documentId)
+            {
+                if (documentId == this.lastDocumentId)
+                {
+                    return;
+                }
+
+                this.lastDocumentId = documentId;
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+                if (document is null)
+                {
+                    symbolTree.ItemsSource = new List<DocSymbol>();
+                    return;
+                }
+
+                document.TryGetText(out var text);
+                if (text is null)
+                {
+                    symbolTree.ItemsSource = new List<DocSymbol>();
+                    return;
+                }
+
+                var textBuffer = text.Container.GetTextBuffer();
+                this.snapshot = textBuffer.CurrentSnapshot;
+                var isCorrectType = textBuffer.ContentType.IsOfType(ContentTypeNames.RoslynContentType);
```

Comment:
Is this check still needed? I don't think we should be called unless its a C# or VB document

Reply:
Since ActiveDocumentChanged is called on several document types, I updated this check and added a comment.